lib/DDG/Publisher/Site/Duckduckhack/Root.pm

```diff
@@ -2,344 +2,13 @@ package DDG::Publisher::Site::Duckduckhack::Root;
 
 use MooX;
 
-use IO::All;
-use Data::Dumper;
-use DDP;
-
-use File::Path qw(make_path);
-use File::Find;
-use File::chdir;
-
-use Path::Class;
-
-use Markdent::Handler::HTMLStream::Document;
-use Markdent::Parser;
-
-use Text::Trim;
-
-use JSON;
-
-use IPC::Run qw{run timeout};
-
 with qw(
 	DDG::Publisher::DirRole
 );
 
-has nav_file => (
-	is => 'ro',
-	lazy => 1,
-	builder => sub {'ddh-index.md'},
-);
-
-has prev_next_file => (
-	is => 'ro',
-	lazy => 1,
-	builder => sub {'ddh-prev-next.json'},
-);
-
-has source_dir => (
-	is => 'ro',
-	lazy => 1,
-	builder => 1,
-);
-
-sub _build_source_dir {
-	my ( $self ) = @_;
-
-	my $cache_dir = $self->site->publisher->cache_dir;
-
-	my $source_dir = dir($cache_dir,'duckduckgo-documentation/duckduckhack');
-
-	{
-		my ($in, $out, $err);    
-
-		#    die $cache_dir;
-		if (-d $source_dir) {
-			local $CWD = $source_dir;
-			run [
-				'git', 'pull'
-			], \$in, \$out, \$err, timeout(60) or die "$err (error $?) $out";
-		} else {
-			local $CWD = $cache_dir;
-			run [
-				'git', 'clone', 'https://github.com/duckduckgo/duckduckgo-documentation.git'
-			], \$in, \$out, \$err, timeout(60) or die "$err (error $?) $out";
-		}
-	}
-
-#	warn qq(DUCKDUCKHACK source_dir: $source_dir\n);
-#	die;
-
-	return $source_dir;
-}
-
 sub path { '/' }
 
-# Builds the prev/next hash to push to each template.
-sub get_prev_next {
-    my $self = shift;
-
-    my $prev_next_path = dir($self->source_dir,$self->prev_next_file);
-    my $json = io($prev_next_path)->slurp;    
-    $json =~ s/\.md//g;
-    my $data = decode_json $json;
-
-#    die p($data);
-
-    return $data;
-}
-
-# Builds the nav hash to push to each template.
-sub get_nav {
-    my $self = shift;
-
-    my $nav_path = dir($self->source_dir,$self->nav_file);
-    my $markdown = io($nav_path)->slurp;    
-
-#    die $markdown;
-
-    my %nav = ();
-    my @nav = ();
-    HEADING: foreach my $heading (split(/\n\-/s,$markdown)) {
-	next HEADING if !$heading;
-	next HEADING if $heading !~ /\*\*/;
-
-	my $title = '';
-	my @sec = ();
-      LINE: foreach my $line (split(/\n/,$heading)) {
-	  if ($line =~ /\*\*([^\*]+)/) {
-	      $title = $1;
-
-	  # Capturing markdown link.
-	  } elsif ($line =~ /\[([^\]]+)\]\(([^\)]+)/) {
-	      my $section = $1;
-	      my $link = $2;
-	      $link =~ s/^.*\/(.*)\.md/$1/;
-	      my %sec = ();
-	      $sec{'title'} = $section;
-	      $sec{'link'} = $link;
-	      push(@sec,\%sec);
-	      
-	      $nav{$link} = {
-		  'category' => $title,
-		  'title' => $section,
-	      }
-	  }
-      }
-
-	if ($title) {
-	    my %sec = ();
-	    $sec{'title'} = $title;
-	    $sec{'sections'} = \@sec;
-	    push(@nav,\%sec);
-	}
-    }
-
-#    die p(%nav);
-    return \@nav, \%nav;
-}
-
-# For just testing output.
-my $dir_output = '/usr/local/ddg/www-static/duckduckhack.com/tmp';
-
-sub pages {
-	return {} if $ENV{DDH_SKIP_DOCS_BUILD};
-	my $self = shift;
-
-	my %pages = ();
-
-	# Old index.
-	$pages{'index'} = sub {
```

Comment:
@jbarrett this stuff we still need I beleive -- it's really silly, but this file contains the code to build duckduckhack.com **and** the docs.

So `duckduckhack` should now only refer to the static page -- unless we're not handling that in publisher anymore either?